packages/common/src/services/explore/Explore.ts

```diff
@@ -97,18 +103,23 @@ export class Explore {
       const tracks = lineupItems.filter(
         (lineupItem): lineupItem is UserTrack => 'track_id' in lineupItem
       )
+      const { data, signature } = await this.audiusBackendInstance.signDiscoveryNodeRequest()
+      const history = await sdk.full.users.getUsersTrackHistory({
+        id: encodeHashId(currentUserId),
+        encodedDataMessage: data,
+        encodedDataSignature: signature,
+        limit: 100
+      })
+      const activityData = history.data as APIActivityV2[]
+      const listenedToTracks = activityData.map(responseAdapter.makeActivity)
+        .filter(removeNullable) as UserTrackMetadata[]
 
       // Imperfect solution. Ideally we use an endpoint that gives us true/false
       // if a user has listened to a passed in array of tracks.
-      const history = await this.apiClient.getUserTrackHistory({
-        currentUserId,
-        userId: currentUserId,
-        limit: 50
-      })
-      const listens = history.map((item) => item.track?.track_id)
+      const listenendToTrackIds = listenedToTracks.map((track) => track.track_id)
 
       const notListenedToTracks = tracks.filter(
-        (track) => !listens[track.track_id]
+        (track) => !listenendToTrackIds[track.track_id]
```

Comment:
the response adapter changes trackIds back to ints right? just want to make sure

Reply:
yup!